src/Login/redux/reducers/loginReducer.js

```diff
@@ -0,0 +1,39 @@
+import { loginTypes } from '../types';
+
+const initialState = {
+  token: '',
+  username: '',
+  isloggedIn: false,
+  is_logging_in: false,
+  login_failed: false,
+  isLoading: false,
+};
+
+export default function (state = initialState, { type, payload }) {
+  switch (type) {
+    case loginTypes.LOGIN_SUCCESS:
+      return {
+        ...state,
+        login_success: payload,
+        isloggedIn: true,
+        isLoading: false,
+      };
+    case loginTypes.LOGIN_ERROR:
+      return {
+        ...state,
+        login_errors: payload.message,
+        login_failed: true,
+        is_logging_in: false,
+        isLoading: false,
+      };
+    case loginTypes.IS_LOGGING:
```

Comment:
Where's the test for this?

Reply:
l have added the test for this . Thank you